solidity/contracts/bridge/Bridge.sol

```diff
@@ -109,18 +109,101 @@ contract Bridge is Ownable {
+    struct RedemptionRequest {
+        // ETH address of the redeemer who created the request.
+        address redeemer;
+        // Requested TBTC amount in satoshi.
+        uint64 requestedAmount;
+        // Treasury TBTC fee in satoshi at the moment of request creation.
+        uint64 treasuryFee;
+        // Transaction maximum BTC fee in satoshi at the moment of request
+        // creation.
+        uint64 txMaxFee;
+        // UNIX timestamp the request was created at.
+        uint32 requestedAt;
+    }
+
+    /// @notice Represents an outcome of the redemption Bitcoin transaction
+    ///         outputs processing.
+    struct RedemptionTxOutputsInfo {
+        // Total TBTC value in satoshi that should be burned by the Bridge.
+        // It includes the total amount of all BTC redeemed in the transaction 
+        // and the fee paid to BTC miners for the redemption transaction.
+        uint64 totalBurnableValue;
+        // Total TBTC value in satoshi that should be transferred to
+        // the treasury. It is a sum of all treasury fees paid by all
+        // redeemers included in the redemption transaction.
+        uint64 totalTreasuryFee;
+        // Index of the change output. The change output becomes
+        // the new main wallet's UTXO.
+        uint32 changeIndex;
+        // Value in satoshi of the change output.
+        uint64 changeValue;
+    }
+
+    /// @notice Represents wallet state:
+    ///         - Unknown: the wallet is unknown to the Bridge
+    ///         - Active: the wallet can sweep deposits and accept redemption
+    ///           requests
+    ///         - MovingFunds: the wallet was deemed unhealthy and is only
+    ///           expected to move their outstanding funds to another wallet.
+    ///           No other actions are possible.
+    ///         - Closed: the wallet cannot perform any action
+    enum WalletState {Unknown, Active, MovingFunds, Closed}
+
+    /// @notice Holds information about a wallet.
+    struct Wallet {
+        // Current state of the wallet.
+        WalletState state;
+        // The total redeemable value of pending redemption requests targeting
+        // that wallet.
+        uint64 pendingRedemptionsValue;
+    }
+
     /// @notice The number of confirmations on the Bitcoin chain required to
     ///         successfully evaluate an SPV proof.
     uint256 public immutable txProofDifficultyFactor;
 
-    // TODO: Revisit whether it should be updatable or not.
+    /// TODO: Revisit whether it should be governable or not.
     /// @notice Address of the Bank this Bridge belongs to.
     Bank public immutable bank;
 
-    /// TODO: Make it updatable.
+    /// TODO: Make it governable.
     /// @notice Handle to the Bitcoin relay.
     IRelay public immutable relay;
 
+    /// TODO: Revisit whether it should be governable or not.
+    /// @notice Address where the redemptions treasury fees will be sent to.
+    address public immutable treasury;
```

Comment:
Would be great if we could describe what this `treasury` component is (not necessarily in this PR).

Reply:
See dbb7bf37